doc/source/user/absolute_beginners.rst

```diff
@@ -0,0 +1,1913 @@
+
+If you aren't familiar with this style, it's very easy to understand. 
+If you see ``>>>``, you're looking at **input**, or the code that 
+you would enter. Everything that doesn't have ``>>>`` in front of it 
+is **output**, or the results of running your code.
+
+
+What’s the difference between a Python list and a NumPy array? 
+--------------------------------------------------------------
+  
+NumPy gives you an enormous range of fast and efficient ways of creating arrays
+and manipulating numerical data inside them. While a Python list can contain
+different data types within a single list, all of the elements in a NumPy array
+should be homogenous. The mathematical operations that are meant to be performed
+on arrays would be extremely inefficient if the arrays weren't homogenous. 
+
+**Why use NumPy?**
+
+NumPy arrays are faster and more compact than Python lists. An array consumes
+less memory and is convenient to use. NumPy uses much less memory to store data
+and it provides a mechanism of specifying the data types. This allows the code
+to be optimized even further. 
+
+What is an array? 
+-----------------
+
+An array is a central data structure of the NumPy library. An array is a grid of
+values and it contains information about the raw data, how to locate an element,
+and how to interpret an element. It has a grid of elements that can be indexed
+in `various ways <https://numpy.org/devdocs/user/quickstart.html#indexing-slicing-and-iterating>`_.
+The elements are all of the same type, referred to as the array ``dtype``. 
+
+An array can be indexed by a tuple of nonnegative integers, by booleans, by
+another array, or by integers. The ``rank`` of the array is the number of
+dimensions. The ``shape`` of the array is a tuple of integers giving the size of
+the array along each dimension.
+
+One way we can initialize NumPy arrays is from Python lists, using nested lists
+for two- or higher-dimensional data.. 
+
+For example::
+
+  >>> array = np.array([1, 2, 3, 4, 5, 6])
+
+or::
+
+  >>> a = np.array([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]])
+
+We can access the elements in the array using square brackets. When you're
+accessing elements, remember that indexing in NumPy starts at 0. That means that
+if you want to access the first element in your array, you'll be accessing
+element "0".
+
+::
+
+  >>> print(a[0])
+  [1 2 3 4]
+
+
+More information about arrays 
+-----------------------------
+
+::
+
+  1D array, 2D array, ndarray, vector, matrix
+
+------
+
+You might occasionally hear an array referred to as a "ndarray," which is
+shorthand for "N-dimensional array." An N-dimensional array is simply an array
+with any number of dimensions. You might also hear **1-D**, or one-dimensional
+array, **2-D**, or two-dimensional array, and so on. The NumPy ``ndarray`` class
+is used to represent both matrices and vectors. A **vector** is an array with a
+single column, while a **matrix** refers to an array with multiple columns.
```

Comment:
I think we need to be very careful with this explanation.

Columns vs rows are a property of 2d arrays. Are you trying to describe a vector as shape `(n,)` or shape `(n, 1)`?

Reply:
Good point! Do you have a preference is how I explain this? Maybe using "index" rather than "column" would be more accurate, or do you have more thoughts? I want to give a clear and concise overview without getting into the technicalities here, if possible.

Reply:
I don't yet have a great idea how to present this concisely, just wanted to draw attention to this in case someone else does.

Reply:
Great!

Reply:
How about this:

A **vector** is an array with a single dimension (note: there is no difference
between row and column vectors), while a **matrix** refers to an
array with two dimensions.  For **3-D** or higher dimensional arrays, the term
**tensor** is also commonly used.